add current sensor + power sensor variance diff to ILSE docs (#962)
PowerGridModel/power-grid-model: docs/user_manual/calculations.md

- two times the number of nodes with a voltage sensor with magnitude and angle
- two times the number of nodes without appliances connected
- two times the number of nodes where all connected appliances are measured by a power sensor

> - two times the number of branches with a power sensor and/or a current sensor

Reviewer: Maybe citing  https://github.com/PowerGridModel/power-grid-model/pull/962/files#diff-8886678ab74c3ed8d4945b40bf22941b6d68fa0b3dfeef8961447c013a389693R95-R96 at the end. That way it is not convoluted, but it is clearly connected to where this is explained.

> - two times the number of branches with a power sensor and/or a current sensor

Author: confirmed off-line to be good as-is :+1: resolving

```{note}
Having enough measurements does not necessarily mean that the system is observable. The location of the measurements is
also of importance, i.e., the measurements should be topologically independent. Additionally, there should be at least one voltage measurement.
```

```{note}
Global angle current measurements require at least one voltage angle measurement to make sense.

> ```

Reviewer 2: Cany you move this section into the warning mentioned below:
"For this reason, global angle current measurements require at least one voltage angle measurement."

Author: incorporated cfr. offline discussion

Based on the requirements of observability mentioned above, users need to satisfy at least the following conditions for state estimation calculation in `power-grid-model`.

- `n_voltage_sensor >= 1`
- If no voltage phasor sensors are available, then the following conditions should be satisfied:
  - There are no global angle current sensors
  - And:
    - Either: `n_unique_power_or_current_sensor >= n_bus - 1`.

>     - And/or: `n_unique_power_or_current_sensor + n_voltage_sensor_with_phasor >= n_bus`

Reviewer: Commenting like this because suggestions are being funny:

- If no voltage phasor sensors are available, then one of the following conditions should be satisfied:
  - There are no global angle current sensors and `n_unique_power_or_current_sensor >= n_bus - 1`.
  - There are global angle current sensors and `n_unique_power_or_current_sensor + n_voltage_sensor_with_phasor >= n_bus`
  
  or something along those lines. I found the first a bit confusing.

Author: whoops i completely misunderstood the original code. Fixed.

    \end{eqnarray}
$$

- Branch current with global angle: Linear WLS requires a complex current phasor. The global angle current measurement captures
the phase offset relative to the same predetermined reference phase against which the voltage angle is measured. As a result,

> at least one voltage measurement with angle measurement is required in the grid if a global angle current sensor is used.

Reviewer: I would explicitly state that even though usually the defaulted reference angle of the voltage sensor is set to zero if not specified, this is not enough when global current sensors are present. I foresee this being a data problem coming our way later on if we don't over state it as much as possible.

$$
    \begin{eqnarray}
            \underline{I} = I_i \cdot e^{j \theta_i}
    \end{eqnarray}
$$

- Branch current with local angle: Linear WLS requires a complex current phasor. Sometimes, (accurate) voltage measurements are
not available for a branch, which means no power measurement is possible. Using only the current amplitude is difficult.
However, it may be possible to obtain a reasonably accurately measurement of the current amplitude and the relative phase angle
to the voltage. In this way, we still have enough information for the state estimation. The resulting local current phasor

> $\underline{I}_{\text{local}}$ can be translated to the global current phasor (see above) for iterative linear state estimation.

Reviewer: Nitpick: Replace the (see above) for a hyperlink to the global. We don't know when/if this would be shifted around. And with the markdown lint, we would get a heads up if that link is broken.

Reviewer: That is fine, it was just a nitpick in any case. Resolving.

    \end{eqnarray}
$$

> - Branch current with global angle: Linear WLS requires a complex current phasor. The global angle current measurement captures

Reviewer 2: We are repeating : `Linear WLS requires a complex current phasor.` a lot.
We can add a text before these bullet points:

We need to convert all measurements into either complex voltage or complex current for Linear WLS.

Author: attempted to solve it. can you please check whether it's good enough?

the phase offset relative to the same predetermined reference phase against which the voltage angle is measured. It is not
sufficient to use the default zero-phase reference offset, because the reference point is not uniquely determined when there
are no voltage angle measurements, resulting in ambiguities. As a result, using any global angle current sensor in the grid
requires at least one voltage phasor measurement (with angle).

$$
    \begin{eqnarray}
            \underline{I} = I_i \cdot e^{j \theta_i}
    \end{eqnarray}
$$

- Branch current with local angle: Linear WLS requires a complex current phasor. Sometimes, (accurate) voltage measurements are

> not available for a branch, which means no power measurement is possible. Using only the current amplitude is difficult.
> However, it may be possible to obtain a reasonably accurately measurement of the current amplitude and the relative phase angle

Reviewer 2: ```suggestion
not available for a branch, which means no power measurement is possible. 
However, it may be possible to obtain a reasonably accurate measurement of the current amplitude and the relative phase angle (ie. from the power factor)
```

Author: will try to rephrase

Author: Can you please double-check whether it's better now?

    \end{eqnarray}
$$

- Branch current with global angle: The global angle current measurement captures
the phase offset relative to the same predetermined reference phase against which the voltage angle is measured. It is not
sufficient to use the default zero-phase reference offset, because the reference point is not uniquely determined when there
are no voltage angle measurements, resulting in ambiguities. As a result, using any global angle current sensor in the grid
requires at least one voltage phasor measurement (with angle).

$$
    \begin{eqnarray}
            \underline{I} = I_i \cdot e^{j \theta_i}
    \end{eqnarray}
$$

- Branch current with local angle: Sometimes, (accurate) voltage measurements are available for a branch,

> which means no power measurement is possible. While it is not straightforward to use magnitude-only measurements of currents in

Reviewer 2: `Sometimes, (accurate) voltage measurements are available for a branch,
which means no power measurement is possible. `
Not sure if I understand the intent here correctly.

Reviewer 2: Ahh, maybe you meant
```suggestion
- Branch current with local angle: Sometimes, (accurate) voltage measurements are not available for a branch,
which means no power measurement is possible. While it is not straightforward to use magnitude-only measurements of currents in
```
